src/module/actor/SR5Actor.ts

```diff
@@ -125,69 +66,147 @@ export class SR5Actor extends Actor {
+        ['firewall', 'sleaze', 'data_processing', 'firewall'].forEach((key) => {
+            if (matrix[key]) {
+                limits[key] = {
+                    value: matrix[key].value,
+                    base: matrix[key].base,
+                    mod: matrix[key].mod,
+                    hidden: true,
+                };
+                attributes[key] = {
+                    value: matrix[key].value,
+                    base: matrix[key].base,
+                    mod: matrix[key].mod,
+                    hidden: true,
+                };
             }
-            // MODIFIES MATRIX ATTRIBUTES
-            if (item.type === 'device' && itemData.technology?.equipped) {
-                matrix.device = item._id;
-                matrix.condition_monitor.max = itemData.technology.condition_monitor?.max || 0;
-                matrix.condition_monitor.value = itemData.technology.condition_monitor?.value || 0;
-                matrix.rating = itemData.technology.rating;
-                matrix.is_cyberdeck = itemData.category === 'cyberdeck';
-                matrix.name = item.name;
-                matrix.item = itemData;
-
-                if (itemData.category === 'cyberdeck' && itemData.atts) {
-                    for (let [key, att] of Object.entries(itemData.atts)) {
-                        matrix[att.att].value += att.value;
-                        matrix[att.att].device_att = key;
-                    }
-                } else {
-                    matrix.firewall.value += matrix.rating || 0;
-                    matrix.data_processing.value += matrix.rating || 0;
+        });
+    }
+
+    /**
+     * Prepare the armor data for the Item
+     * - will only allow one "Base" armor item to be used
+     * - all "accessories" will be added to the armor
+     * @param data
+     * @param items
+     */
+    static prepareArmor(data: SR5ActorData, items: SR5ItemDataWrapper[]) {
+        const { armor } = data;
+        armor.base = 0;
+        armor.value = 0;
+        armor.mod = {};
+        for (const element of Object.keys(CONFIG.SR5.elementTypes)) {
+            armor[element] = 0;
+        }
+
+        const equippedArmor = items.filter((item) => item.isArmor() && item.isEquipped());
+        equippedArmor?.forEach((item) => {
+            if (item.isArmorAccessory()) {
+                armor.mod[item.getName()] = item.getArmorValue();
+            } // if not a mod, set armor.value to the items value
+            else {
+                armor.base = item.getArmorValue();
+                armor.label = item.getName();
+                for (const element of Object.keys(CONFIG.SR5.elementTypes)) {
+                    armor[element] = item.getArmorElements()[element];
                 }
             }
-        }
+        });
 
-        armor.mod[game.i18n.localize('SR5.Bonus')] = modifiers['armor'];
+        if (data.modifiers['armor']) armor.mod[game.i18n.localize('SR5.Bonus')] = data.modifiers['armor'];
```

Comment:
Curly braces for clarity
```suggestion
        if (data.modifiers['armor']) {
            armor.mod[game.i18n.localize('SR5.Bonus')] = data.modifiers['armor'];
        }
```